tools/execsnoop.py

```diff
@@ -243,15 +253,16 @@ def parse_uid(user):
 b.attach_kretprobe(event=execve_fnname, fn_name="do_ret_sys_execve")
 
 # header
-if args.containersmap:
-    print("%-16s %-16s %-16s %-16s" % ("NODE", "NAMESPACE", "PODNAME", "CONTAINERNAME"), end="")
-if args.time:
-    print("%-9s" % ("TIME"), end="")
-if args.timestamp:
-    print("%-8s" % ("TIME(s)"), end="")
-if args.print_uid:
-    print("%-6s" % ("UID"), end="")
-print("%-16s %-6s %-6s %3s %s" % ("PCOMM", "PID", "PPID", "RET", "ARGS"))
+if not args.json:
+    if args.containersmap:
+        print("%-16s %-16s %-16s %-16s" % ("NODE", "NAMESPACE", "PODNAME", "CONTAINERNAME"), end="")
```

Comment:
Since it will not print headers in case of `--json`, I wonder how it can signal to the caller that the BPF program is installed, in case we want the `kubectl gadget execsnoop` command or an Headlamp plugin to present this information somehow (or an error if the BPF program does not pass the verifier).

In the network policy advisor, we had the event "ready" in json to signal that (see [definition](https://github.com/kinvolk/inspektor-gadget/blob/3720d7b8a4658a3b5ed66f0f1c2cd37bce1547f1/pkg/gadgets/networkpolicy/types/event.go#L18) and [usage](https://github.com/kinvolk/inspektor-gadget/blob/3720d7b8a4658a3b5ed66f0f1c2cd37bce1547f1/gadget-container/gadgets/networkpolicyadvisor/main.go#L213)). It does not feel appropriate to do the same here...

Any opinions on this?

Reply:
I think there are different use cases. If the program fails to pass the verifier (or any other problem) it'll return with an exit code different than 0, that'd allow the caller to understand that something wrong is going on and to inform it to the user. 

What I don't know is how to signal that the program is installed and capturing events, like a synchronization mechanism. Are you aware of a use case where the caller must need to know that the program is installed and running for sure?

Reply:
My use case is that I want to be able to catch the first events of a pod, so as a user, I would start execsnoop, wait until it tells me the tracer is installed, then create the pod that I want to to inspect. Since bcc tools take a few seconds to compile, users might wrongly assume that it is installed and then start the pod to inspect too early...

So it is useful even if there are no problems with the verifier.

Reply:
I agree with that use case, however I'm not sure how to solve it. If we print anything to the screen any tool parsing the output should understand it. (It can be a json "ready" event or just a "Ready" message). Actually it gets more complicated when we take Inspektor Gadget in consideration. In that case the gadget is ready when it's deployed to **all** nodes. Btw, we don't do it in the current implementation (without json), when the header is printed it means there's a least one gadget running, not that all of them are running. 

I'd propose to open an issue to track it as IMHO it's hard to solve now.

Reply:
I agree. It can be a different issue and it does not have to be fixed in this PR.